app/src/main/java/com/jxc/app/services/UserServiceImp.java

```diff
@@ -0,0 +1,35 @@
+package com.jxc.app.services;
+
+import com.jxc.app.exceptions.UserNotFoundException;
+import com.jxc.app.models.User;
+import com.jxc.app.repositories.UserRepository;
+
+import java.util.List;
+import java.util.Optional;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+@Service
+public class UserServiceImp implements UserService{
+
+    @Autowired
+    UserRepository userRepository;
+
+    public User save(User user){
+        return userRepository.save(user);
+    }
+
+    public User findUserById(String id){
+        final Optional<User> optionalUser = this.userRepository.findById(id);
+        if (!optionalUser.isPresent()){
+            throw new UserNotFoundException("User with id : "+ id + " does not exist.");
+        }
+        return optionalUser.get();
+    }
```

Comment:
Désolé de dire ça mais j'ai vomi aha ! 

Lis ça :). https://dzone.com/articles/using-optional-correctly-is-not-optional

```suggestion
    public User findUserById(String id){
        return this.userRepository.findById(id)
            .orElseThrow(() ➜ new UserNotFoundException("User with id : "+ id + " does not exist."));
    }
```